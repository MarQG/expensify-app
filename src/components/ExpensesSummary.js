import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral  from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


export const ExpensesSummary = ({ expenseCount, expensesTotal, expensesCountTotal }) => {
    const expensesWord = expenseCount === 1 ? 'expense' : 'expenses';
    const expensesCountWord = expensesCountTotal === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Showing <span>{expenseCount}</span> of {expensesWord} totalling <span>{formattedExpensesTotal}</span> </h1>
                <h1 className="page-header__title"><span>{expensesCountTotal}</span> hidden {expensesCountWord}  not being shown.</h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    const totalExpensesCount = state.expenses.length - visibleExpenses.length;
    return{
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        expensesCountTotal: totalExpensesCount
    }
};

export default connect(mapStateToProps)(ExpensesSummary);