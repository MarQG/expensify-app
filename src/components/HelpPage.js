import React from 'react';

const HelpPage = () => (
    <div>
        This from my Help Component
    </div>
);

export default HelpPage;