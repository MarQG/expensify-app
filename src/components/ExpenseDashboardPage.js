import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
import Header from '../components/Header';

const ExpenseDashboardPage = () => (
    <div>
        <Header />
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;