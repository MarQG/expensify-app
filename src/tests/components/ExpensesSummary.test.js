import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={20035}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses summary with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={120035}/>);
    expect(wrapper).toMatchSnapshot();
});