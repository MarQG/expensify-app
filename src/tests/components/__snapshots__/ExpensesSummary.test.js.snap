// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with multiple expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Showing 
      <span>
        3
      </span>
       of 
      expenses
       totalling 
      <span>
        $1,200.35
      </span>
       
    </h1>
    <h1
      className="page-header__title"
    >
      <span />
       hidden 
      expenses
        not being shown.
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render expenses summary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Showing 
      <span>
        1
      </span>
       of 
      expense
       totalling 
      <span>
        $200.35
      </span>
       
    </h1>
    <h1
      className="page-header__title"
    >
      <span />
       hidden 
      expenses
        not being shown.
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
